kythe/go/extractors/gcp/config/gradle.go

```diff
@@ -27,40 +27,50 @@ import (
 	"google.golang.org/api/cloudbuild/v1"
 )
 
-func gradleStep(conf *rpb.ExtractionHint) *cloudbuild.BuildStep {
-	return &cloudbuild.BuildStep{
-		Name: constants.GCRGradleImage,
-		Args: []string{
-			"clean",
-			// TODO(#3126): If compile-test has to be done as a separate
-			// step, then we also need to fix this in the same way as we do
-			// for multiple repo support.  Probably this would need to be
-			// done with multiple steps (but making sure to not clobber
-			// output).
-			// The alternative here is to fall back to using clean install,
-			// which should also work.
-			"build",
-			"-s", // Prints stacktraces for user exceptions.
-			"-S", // Prints verbose stacktraces.
-			"-d", // Logs in debug mode.
-			"-b",
-			path.Join(codeDirectory, conf.Root, "build.gradle"),
-		},
-		Volumes: []*cloudbuild.Volume{
-			&cloudbuild.Volume{
-				Name: javaVolumeName,
-				Path: constants.DefaultExtractorsDir,
+type gradleStepper struct{}
+
+// artifactsPre implements part of stepper
+func (m gradleStepper) artifactsPre() []string {
+	return []string{path.Join(outputDirectory, "javac-extractor.err")}
+}
+
+func (m gradleStepper) steps(conf *rpb.ExtractionHint) []*cloudbuild.BuildStep {
```

Comment:
If you are going to add a impl comment for artifactsPre, you might as well add one here as well.